src/Core/src/Platform/iOS/MauiView.cs

```diff
@@ -122,12 +139,8 @@ public override void LayoutSubviews()
 			var widthConstraint = bounds.Width;
 			var heightConstraint = bounds.Height;
 
-			// If the SuperView is a MauiView (backing a cross-platform ContentView or Layout), then measurement
-			// has already happened via SizeThatFits and doesn't need to be repeated in LayoutSubviews. But we
-			// _do_ need LayoutSubviews to make a measurement pass if the parent is something else (for example,
-			// the window); there's no guarantee that SizeThatFits has been called in that case.
```

Comment:
I'm not able to confirm the accuracy of this statement.

- Currently SizeThatFits doesn't take into account the safearea so the parent SizeThatFits call might possibly being using an inaccurate size measurement
- I realize in this PR I've added a check for safearea inside `SizeThatFits`, but, the SafeAreaInsets aren't set on a UIView until the first time we reach `LayoutSubViews` so, on initial load the `SizeThatFits` calls are going to be using the wrong layout sizing, so, if we get here and the bounds have changed then we need to rerun the measure pass.

Or I'm wrong :-) 

The problem we're seeing is that if you have a layout like this

```XAML
    <HorizontalStackLayout>
        <Button Text="Button with a longer text to test how the truncating works and see if the image is centered" 
                ImageSource="dotnet_bot_resized.png" Background="LightPink"
                ContentLayout="Top, 10" WidthRequest="200"/>
    </HorizontalStackLayout>
```

The SizeThatFits call will use the size of the screen
And then once we hit `LayoutSubViews` the bounds gets set to (screenheight - safearea) and then we don't redo a measure pass. This leads to the Layout doing an arrange on an improper measure